examples/run_nlp_trainer.py

```diff
@@ -0,0 +1,350 @@
+            ft_hparams.save_folder = f'{task}'
+            ft_hparams.save_num_checkpoints_to_keep = 0
+            save_artifact_name = f'{{run_name}}/{task}/ep{{epoch}}-ba{{batch}}-rank{{rank}}'
+            save_latest_artifact_name = f'{{run_name}}/{task}/latest-rank{{rank}}'
+            ft_hparams.save_artifact_name = save_artifact_name
+            ft_hparams.save_latest_artifact_name = save_latest_artifact_name
+            ft_hparams.save_overwrite = True
+
+    print(f" --------\n SPAWNING NEW TASK: {task}\n DEVICE: {torch.cuda.current_device()}\n --------")
+
+    trainer = ft_hparams.initialize_object()
+    trainer.fit(duration='3ep')
+    print(f"\nFINISHED TRAINING TASK {task.upper()}\n")
+    return trainer.state.current_metrics
+
+'''Get the run name by pre-constructing a dummy TrainerHparams instance. ''' 
+def get_run_name():
+    hp = TrainerHparams.create(cli_args=['--file', 'composer/yamls/models/bert-base.yaml'])
+    return hp.run_name
+
+''' Validate NLP specific entrypoint arguments. ''' 
+def validate_args(args) -> None:
+    if args.training_scheme != 'pretrain' and args.training_scheme != 'finetune' and args.training_scheme != 'all':
+            raise ValueError('training_scheme must be one of "pretrain," "finetune," or "all"')
+
+    if args.training_scheme == 'finetune' and args.load_path is None: 
+        raise ValueError('load_path to checkpoint folder must be specified if finetuning a model')
+
+    if not args.save_locally and args.bucket is None:
+        raise ValueError('S3 bucket name must be specified if saving to cloud')
+
+''' Get TrainerHparams arguments from CLI and add NLP entrypoint specific arguments. ''' 
+def get_args():
+    parser = hp.get_argparse(TrainerHparams)
+    parser.add_argument('--training_scheme', help='training scheme used (one of "pretrain", "finetune", or "all")')
+    parser.add_argument('--save_locally', action=argparse.BooleanOptionalAction, help='save to local directory or to cloud using object store if --no-save-locally is enabled')
+    parser.add_argument('--bucket', help='S3 bucket name to pull from/save to if --no-save_locally is enabled')
+    
+    args = parser.parse_args()
+    validate_args(args)
+    return args
+
+''' Logic for handling a pretraining job spawn based on storage and training settings.'''
+def run_pretrainer(args):
+    root_dir = os.path.join(os.path.dirname(composer.__file__), '..')
+    training_script = os.path.join(root_dir, 'examples/run_composer_trainer.py')
+    if args.save_locally:
+        args.save_folder = os.path.join(args.run_name, 'checkpoints')
+        if args.training_scheme == 'all':
+            new_save_folder = os.path.join(args.save_folder, 'pretrained')
+        else: # pretrain only
+            new_save_folder = args.save_folder
+        
+        proc = subprocess.Popen(['composer', training_script, '-f', f"{args.file}", '--save_folder', f"{new_save_folder}", 
```

Comment:
Can you create a jira to add a functional api to the composer launch script? This works but not a big fan of calling python via a subprocess. But there's no way around this for no so it's fine.